### ISSUE TITLE
Stochastic weight averaging without the learning rate scheduler

### ISSUE THREAD

**Issue by davidgilbertson** [Lightning-AI/lightning#15916]
## 🚀 Feature

The ability to use stochastic weight averaging _without_ the SWALR learning rate scheduler.

### Motivation

Unless I'm mistaken, the current implementation of `StochasticWeightAveraging` will do the averaging, anneal the learning rate to a constant value, and update the batch norms. This is wonderful, but what if I want to experiment with a different learning rate schedule while SWA is active (as they do in the paper, using Cyclic learning rates for some models).

### Pitch

It would be nice to have a parameter on the `StochasticWeightAveraging` callback, `anneal_to_constant_lr` that defaults to `True` (resulting in the current behaviour) but that I can set to `False` so that whatever learning rate schedules I have in place are not affected by the use of this callback (but the averaging, batch norm, and switching which model is used for the validation/test step all stay as is).

### Alternatives

I'm new to Lightning, but assume I can implement the underlying PyTorch SWA components to get this effect, I just haven't worked out how yet and was hoping I could use this callback.

**Comment by c3-utsavdutta98** [comment]
Any updates?
